sys/linux/damon.txt

```diff
@@ -0,0 +1,97 @@
+openat$damon_attrs(fd const[AT_FDCWD], file ptr[in, string["/sys/kernel/debug/damon/attrs"]], flags flags[open_flags], mode flags[open_mode]) fd_damon_attrs
+openat$damon_init_regions(fd const[AT_FDCWD], file ptr[in, string["/sys/kernel/debug/damon/init_regions"]], flags flags[open_flags], mode flags[open_mode]) fd_damon_init_regions
+openat$damon_monitor_on(fd const[AT_FDCWD], file ptr[in, string["/sys/kernel/debug/damon/monitor_on"]], flags flags[open_flags], mode flags[open_mode]) fd_damon_monitor_on
+openat$damon_schemes(fd const[AT_FDCWD], file ptr[in, string["/sys/kernel/debug/damon/schemes"]], flags flags[open_flags], mode flags[open_mode]) fd_damon_schemes
+openat$damon_kdamond_pid(fd const[AT_FDCWD], file ptr[in, string["/sys/kernel/debug/damon/kdamond_pid"]], flags flags[open_flags], mode flags[open_mode]) fd_damon_kdamond_pid
+openat$damon_mk_contexts(fd const[AT_FDCWD], file ptr[in, string["/sys/kernel/debug/damon/mk_contexts"]], flags flags[open_flags], mode flags[open_mode]) fd_damon_mk_contexts
+openat$damon_rm_contexts(fd const[AT_FDCWD], file ptr[in, string["/sys/kernel/debug/damon/rm_contexts"]], flags flags[open_flags], mode flags[open_mode]) fd_damon_rm_contexts
+
+write$damon_target_ids(fd fd_damon_target_ids, buf ptr[in, damon_target_id_input], count len[buf])
+write$damon_attrs(fd fd_damon_attrs, buf ptr[in, damon_attrs], count len[buf])
+write$damon_init_regions(fd fd_damon_init_regions, buf ptr[in, damon_init_regions], count len[buf])
+write$damon_monitor_on(fd fd_damon_monitor_on, buf ptr[in, string["on"]], count len[buf])
+write$damon_schemes(fd fd_damon_schemes, buf ptr[in, damon_schemes], count len[buf])
+write$damon_mk_contexts(fd fd_damon_mk_contexts, buf ptr[in, damon_contexts], count len[buf])
+write$damon_rm_contexts(fd fd_damon_rm_contexts, buf ptr[in, damon_contexts], count len[buf])
+
+read$damon_target_ids(fd fd_damon_target_ids, buf ptr[out, array[int8]], count len[buf])
+read$damon_init_regions(fd fd_damon_init_regions, buf ptr[out, array[int8]], count len[buf])
+read$damon_attrs(fd fd_damon_attrs, buf ptr[out, array[int8]], count len[buf])
+read$damon_monitor_on(fd fd_damon_monitor_on, buf ptr[out, array[int8]], count len[buf])
+read$damon_schemes(fd fd_damon_schemes, buf ptr[out, array[int8]], count len[buf])
+read$damon_kdamond_pid(fd fd_damon_kdamond_pid, buf ptr[out, array[int8]], count len[buf])
+read$damon_mk_contexts(fd fd_damon_mk_contexts, buf ptr[in, array[string]], count len[buf])
+read$damon_rm_contexts(fd fd_damon_rm_contexts, buf ptr[in, array[string]], count len[buf])
+
+close$damon_target_ids(fd fd_damon_target_ids)
+close$damon_init_regions(fd fd_damon_init_regions)
+close$damon_attrs(fd fd_damon_attrs)
+close$damon_monitor_on(fd fd_damon_monitor_on)
+close$damon_schemes(fd fd_damon_schemes)
+close$damon_mk_contexts(fd fd_damon_mk_contexts)
+close$damon_rm_contexts(fd fd_damon_rm_contexts)
+
+damon_schemes {
```

Comment:
It seems that kernel accepts an ascii string here:
https://elixir.bootlin.com/linux/v5.16-rc1/source/mm/damon/dbgfs.c#L203
If so you need to from a string using spaces and fmt type.
For fmt type see:
https://github.com/google/syzkaller/blob/master/docs/syscall_descriptions_syntax.md

Currently this writes a binary struct with 4-byte ints.

Reply:
Thank you for the tip!